opa/datadog_checks/opa/opa.py

```diff
@@ -0,0 +1,72 @@
+# (C) Datadog, Inc. 2020
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from datadog_checks.base import ConfigurationError, OpenMetricsBaseCheck
+
+METRIC_MAP = {
+    'http_request_duration_seconds': 'request.duration',
+}
+
+
+class OpaCheck(OpenMetricsBaseCheck):
+    DEFAULT_METRIC_LIMIT = 0
+
+    def __init__(self, name, init_config, instances=None):
+        default_instances = {
+            'opa': {
+                'metrics': [METRIC_MAP],
+                'send_distribution_sums_as_monotonic': 'true',
+                'send_distribution_counts_as_monotonic': 'true',
+            }
+        }
+
+        super(OpaCheck, self).__init__(
+            name, init_config, instances, default_instances=default_instances, default_namespace='opa'
+        )
+
+    def _http_check(self, url, check_name):
+        try:
+            response = self.http.get(url)
+            response.raise_for_status()
+        except Exception as e:
+            self.service_check(check_name, self.CRITICAL, message=str(e))
+        else:
+            if response.status_code == 200:
+                self.service_check(check_name, self.OK)
+            else:
+                self.service_check(check_name, self.WARNING)
+
+    def _get_policies(self, opa_url):
+
+        policies_url = opa_url + "/v1/policies"
+        try:
+            response = self.http.get(policies_url)
+            policies = response.json()
+        except Exception:
+            pass
+        else:
+            self.gauge('opa.policies', len(policies['result']), tags=[])
+
+    def check(self, instance):
+        opa_url = instance.get("opa_url")
+        if opa_url is None:
+            raise ConfigurationError("Each instance must have a url to the opa service")
+
+        prometheus_url = instance.get("prometheus_url")
+        if prometheus_url is None:
+            raise ConfigurationError("Each instance must have a url to the prometheus endpoint")
+
+        health_url = opa_url + "/health"
+        plugins_url = health_url + "?plugins"
+        bundles_url = health_url + "?bundles"
+
+        # General service health
+        self._http_check(health_url, 'opa.health')
```

Comment:
In the `service_checks.json` file this service check is named `opa.prometheus.health`.
Note that the `OpenMetricsCheckBase.check` method will submit `opa.prometheus.health` by default, so you can probably remove this line.

Reply:
Those are different checks. The `opa.prometheus.health` is the prometheus endpoint (added as you said by the base check), the `opa.health` one checks the health endpoint (not the metrics endpoint), which is different

Reply:
Oh I see, thanks for the clarification 👍